src/modules/hyprland/workspaces.cpp

```diff
@@ -181,6 +181,9 @@ auto Workspaces::update() -> void {
   }
 
   for (auto &workspace : m_workspaces) {
+    auto clients_data = gIPC->getSocket1JsonReply("clients");
```

Comment:
Shouldn't you get the clients_data outside of the for-loop? Right now you're getting the data every iteration of the loop, does the data change in between iterations?

Reply:
Good catch, pasted it inside loop accidentally.